src/dp.rs

```diff
@@ -1,16 +1,76 @@
 // SPDX-License-Identifier: MPL-2.0
 
 //! Differential privacy (DP) primitives.
-use std::fmt::Debug;
+//!
+//! There are three main traits defined in this module:
+//!
+//!  - `DifferentialPrivacyBudget`: Implementors should be types of budgets for DP-mechanisms.
+//!    Examples: zCDP, ApproximateDP (Epsilon-Delta), PureDP
+//!
+//!  - `DifferentialPrivacyDistribution`: Distribution from which noise is sampled.
+//!    Examples: DiscreteGaussian, DiscreteLaplace
+//!
+//!  - `DifferentialPrivacyStrategy`: This is a combination of choices for budget and distribution.
+//!    Examples: zCDP-DiscreteGaussian, EpsilonDelta-DiscreteGaussian
+//!
+use num_bigint::{BigInt, BigUint, TryFromBigIntError};
+use num_rational::{BigRational, Ratio};
 
-/// Positive rational number to represent DP and noise distribution parameters in protocol messages
-/// and manipulate them without rounding errors.
+/// Errors propagated by methods in this module.
+#[derive(Debug, thiserror::Error)]
+pub enum DpError {
+    /// Tried to use an invalid float as privacy parameter.
+    #[error(
+        "DP error: input value was not a valid privacy parameter. \
+             It should to be a non-negative, finite float."
+    )]
+    InvalidFloat,
+
+    /// Tried to construct a rational number with zero denominator.
+    #[error("DP error: input denominator was zero.")]
+    ZeroDenominator,
+
+    /// Tried to convert BigInt into something incompatible.
+    #[error("DP error: {0}")]
+    BigIntConversion(#[from] TryFromBigIntError<BigInt>),
+}
+
+/// Positive finite precision rational number to represent DP and noise distribution parameters in
```

Comment:
nit: I think "arbitrary precision" would be more clear/accurate than "finite precision" here
```suggestion
/// Positive arbitrary precision rational number to represent DP and noise distribution parameters in
```